clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -335,9 +336,6 @@ public AlterConfigsResult alterConfigs(Map<ConfigResource, Config> configs) {
     /**
      * Update the configuration for the specified resources with the default options.
      *
-     * Updates are not transactional so they may succeed for some resources while fail for others. The configs for
-     * a particular resource are updated atomically.
```

Comment:
Why do we remove this?

Reply:
Originally I added a similar comment to the new API. @becketqin commented that this is unnecessary. I agree with @becketqin this is unnecessary because `AlterConfigsResult` returns `Map<ConfigResource, KafkaFuture<Void>>` which suggests that some configs may be updated successfully while others fail. Does this make sense?

Reply:
While I understand the sentiment, we have to remember that these APIs will be used by people who are not familiar with Kafka in the same way we are. We often get support questions because things that seem obvious to us are not clear to users. I disagree that we should remove clarification comments like the above in public APIs.

Reply:
@ijuma I see. Sure, I can submit a minor patch tomorrow for this (or please feel free to just commit a minor patch if you prefer). To keep the API document consistent and for the same reason you described, maybe we should have this comment (i.e. API is not transactional) for all those APIs in the AdminClient which may partially succeed, e.g. deleteTopics?

Reply:
Yes, I think that's a good idea. If you are OK with submitting a PR, that would be great. It's not urgent, we should aim to do it before 1.0.0 is released.